app/models/miq_queue_worker_base/runner.rb

```diff
@@ -6,13 +6,24 @@ def after_sync_config
   end
 
   def sync_dequeue_method
+    previous_dequeue_method = @dequeue_method
     @dequeue_method = (worker_settings[:dequeue_method] || :sql).to_sym
+
+    dequeue_method_changed(previous_dequeue_method) if @dequeue_method != previous_dequeue_method
   end
 
   def dequeue_method_via_drb?
     @dequeue_method == :drb && drb_dequeue_available?
   end
 
+  def dequeue_method_via_miq_messaging?
+    @dequeue_method == :miq_messaging && miq_messaging_dequeue_available?
+  end
+
+  def dequeue_method_changed(previous_dequeue_method)
+    @listener_thread&.kill if previous_dequeue_method == :miq_messaging
+  end
```

Comment:
I'm fine with keeping this for now, but I wonder if it's just simpler to kill the worker entirely when the messaging model changes - I assume it's incredibly rare one would change part way through.

Reply:
Yeah we can do that, unclear what changing this while the system is running would mean though since there is likely to be a backlog of messages in the old method so technically we would want to
1. Stop entering new messages to the old queue method
2. Process all existing messages in the queue
3. Restart the worker (or start listening with the new method)

Changing the dequeue method is pretty complicated

Reply:
Yeah, that's true - I guess we would lose those messages that are in the in-memory queue? or would they come back on the next worker?

Reply:
If we were changing e.g. changing the dequeue_method from miq_queue -> kafka but there were 10 messages in the miq_queue those wouldn't be processed until the dequeue_method was changed back unless we process all 10 before actually switching.

This doesn't appear to be handled currently (drb -> sql or vice versa) but that might be because they both come from the miq_queue table eventually.